lib/plaid/user.rb

```diff
@@ -332,11 +334,17 @@ def delete
     # Returns another User record with the same access token, but tied to the
     # new product.
     def upgrade(product)
+      User.upgrade product, access_token, client: client
+    end
+
+    # Internal: Upgrade a User instance.
+    def self.upgrade(product, access_token, client: nil)
```

Comment:
It's used in `User#upgrade` (the instance method), 4 lines up.

`User#upgrade` used to call User.new directly, which is now private.  With this additional layer, while it's still possible to bypass `User#upgrade` and call `User.upgrade` (the class method), the resulting User would be in a valid state, as opposed to if it were possible to call `User.new`, which would result in a User in an invalid state.